File: day1.cpp
#include <iostream>
#include <cassert>

/***
@Question 
* Given a string, return the first recurring character 
* in it,or null if there is no recurring character.
* For example, given the string "acbbac", return "b".
* Given the string "abcdef", return null.
 ***/
using namespace std;

char recurring(string word){
    char characters[word.length()];

    for(int i=0; i< word.length(); i++){
       for(int j =0; j < i; j++){
           if(word[i] == characters[j]){
               return word[i];
           } 
       }characters[i] = word[i];
    }
    return '\0';
}

int main(){
     assert('b' == recurring("abbaa"));
     assert('\0' == recurring("ab"));
}

File: day1.py
"""
@Question 
Given a string, return the first recurring character 
in it,or null if there is no recurring character.
For example, given the string "acbbac", return "b".
Given the string "abcdef", return null.
"""

def recurring(string):
    checker = []

    for char in string:
        if(char in checker):
            return char
        else:
             checker.append(char)
    return None

# Testing 

assert recurring("abcb") == "b", "Error"
assert recurring("abc") ==  None, "Error"
assert recurring("aabcb") == "a", "Error"

